### PR TITLE
Reference list of status chars in borg recreate --filter description

### LINKED ISSUE #2730: Document valid status chars for --filter
The supported status chars should be listed with the help for `borg create --filter` (`--filter STATUSCHARS`).
This could just list the available choices (and validate them), and then link to some more specifc documentation/list, where each of them is explained / listed with a short desc, similar to `borg help pattern`?!

### PR DESCRIPTION
**Pull request by milkey-mouse**
Fixes #2730

### CODE REVIEW

**Comment by milkey-mouse** [comment]
On second thought I'm not sure if it's better to make the list of status chars the metavar like I did or just put them at the end of the helptext...

File: src/borg/archiver.py [hunk 1]
```diff
@@ -2857,7 +2857,7 @@ def define_archive_filters_group(subparser, *, sort_by=True, first_last=True):
                                help='print statistics for the created archive')
         subparser.add_argument('--list', dest='output_list', action='store_true',
                                help='output verbose list of items (files, dirs, ...)')
-        subparser.add_argument('--filter', metavar='STATUSCHARS', dest='output_filter',
+        subparser.add_argument('--filter', metavar='AMUEdbchsfi-x', dest='output_filter',
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: hmm, is this really an improvement or just another place that would need to get updated and could easily be forgotten?

the stuff is already described in 2819+.

File: src/borg/archiver.py [hunk 2]
```diff
@@ -3469,8 +3469,8 @@ def define_archive_filters_group(subparser, *, sort_by=True, first_last=True):
         subparser.set_defaults(func=self.do_recreate)
         subparser.add_argument('--list', dest='output_list', action='store_true',
                                help='output verbose list of items (files, dirs, ...)')
-        subparser.add_argument('--filter', dest='output_filter', metavar='STATUSCHARS',
-                               help='only display items with the given status characters')
+        subparser.add_argument('--filter', metavar='AMUEdbchsfi-x', dest='output_filter',
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: same here

File: src/borg/archiver.py (hunk 1 above)
Comment (milkey-mouse) [inline src/borg/archiver.py]: that was my concern, should I just list them at the end of the help text?

Comment (milkey-mouse) [inline src/borg/archiver.py]: Or should I consolidate both texts, like with a dict with each status char & its meaning that could be pretty printed in the help text or displayed in short form with `"".join(statuschars.keys())`?

Comment (enkore) [inline src/borg/archiver.py]: I think just saying in the --filter help that this is described below is sufficient.

File: src/borg/archiver.py [hunk 3]
```diff
@@ -2858,7 +2858,7 @@ def define_archive_filters_group(subparser, *, sort_by=True, first_last=True):
         subparser.add_argument('--list', dest='output_list', action='store_true',
                                help='output verbose list of items (files, dirs, ...)')
         subparser.add_argument('--filter', metavar='STATUSCHARS', dest='output_filter',
-                               help='only display items with the given status characters (see description)')
+                               help='only display items with the given status characters (listed in borg create --help)')
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py]: as this is the borg create --help text, just keep the "see description" here?

Comment (ThomasWaldmann) [inline src/borg/archiver.py]: also: collapse the changesets into 1.

**Comment by ThomasWaldmann** [comment]
thanks!

**Pull request merged**